### ISSUE TITLE
Dask error when importing pip installed xarray.

### ISSUE THREAD

**Issue by twhughes** [pydata/xarray#5841]
**What happened**:

I am pip installing `xarray` version 0.19.0 on python 3.9.7 and am getting the following error when trying to `import xarray`

```
    import xarray as xr
/usr/local/lib/python3.9/site-packages/xarray/__init__.py:3: in <module>
    from . import testing, tutorial, ufuncs
/usr/local/lib/python3.9/site-packages/xarray/testing.py:8: in <module>
    from xarray.core import duck_array_ops, formatting, utils
/usr/local/lib/python3.9/site-packages/xarray/core/duck_array_ops.py:15: in <module>
    from . import dask_array_compat, dask_array_ops, dtypes, npcompat, nputils
/usr/local/lib/python3.9/site-packages/xarray/core/dask_array_compat.py:5: in <module>
    from .pycompat import dask_version
/usr/local/lib/python3.9/site-packages/xarray/core/pycompat.py:55: in <module>
    dsk = DuckArrayModule("dask")
/usr/local/lib/python3.9/site-packages/xarray/core/pycompat.py:22: in __init__
    duck_array_version = LooseVersion(duck_array_module.__version__)
E   AttributeError: module 'dask' has no attribute '__version__'

```

**What you expected to happen**:

Doing `pip install "xarray[complete]"` fixes the issue.
However, I don't want to install dask or the other optional packages as I want to reduce dependencies in my project.

**Minimal Complete Verifiable Example**:

```python
import xarray  as xr # :)
```

**Anything else we need to know?**:

Thanks

**Environment**:

NA

**Comment by keewis** [comment]
closing as a environment issue, but feel free to ask if you need more help

**Comment by ashwinvis** [comment]
@twhughes @keewis In my case, I encounter this when I:
- Have xarray + dask installed in my environment
- Run any pytest which imports xarray (which in turn imports dask)
- Uninstall dask, but pytest artefacts remain:

```sh
❯ ls venv/lib/python3.9/site-packages/dask/__pycache__
utils_test.cpython-39-pytest-6.2.5.pyc
```

Here are the steps which would *hopefully* reproduce the behaviour:

```py
# Create this small unit test named:
# test_import_xarray.py                                                     
def test_xr():
    import xarray
```

```sh
python -m venv venv
source venv/bin/activate
pip install xarray pytest dask
pytest test_import_xarray.py

pip uninstall dask
ls venv/lib/python3.9/site-packages/dask/__pycache__  # See output above
python -c 'import dask'  # No ImportError!
python -c 'import xarray'
```

Exact versions used Python 3.9.9 along with:
```sh
# requirements.txt
attrs==21.2.0
cloudpickle==2.0.0
dask==2021.11.2
fsspec==2021.11.1
iniconfig==1.1.1
locket==0.2.1
numpy==1.21.4
packaging==21.3
pandas==1.3.4
partd==1.2.0
pluggy==1.0.0
py==1.11.0
pyparsing==3.0.6
pytest==6.2.5
python-dateutil==2.8.2
pytz==2021.3
PyYAML==6.0
six==1.16.0
toml==0.10.2
toolz==0.11.2
xarray==0.20.1
```

**Comment by ashwinvis** [comment]
This in my opinion is a cascade of multiple issues:
- `pytest` creates special bytecode
- `pip` does not clean the whole `__pycache__` directory on uninstall
- `xarray.pycompat.is_duck_dask_array` uses `importlib.import_module('dask')` to determine if it is installed.

**Comment by keewis** [comment]
I don't think `xarray` has anything to do with that issue: if `python -c 'import dask'` succeeds, `xarray` will *have* to assume `dask` is available.

You will probably have to investigate `pip` (why didn't it remove the `__pycache__` directory?) or `pytest` (why was the `pyc` file created in the first place?) to really fix this, but for now you can try setting `PYTHONDONTWRITEBYTECODE` or running `python` with `-B` (for example, `python -B -m pytest`)

**Comment by ashwinvis** [comment]
See also #5236 which could be related which uses `conda`.

I tried to scour through pip and pytest issues, but I can't find a discussion on this. `pip` and `pytest` and `pycache`are too generic keywords. I even found [pytest plugin to remove bytecode but it does not work anymore](https://pypi.org/project/pytest-remove-stale-bytecode/).

I proposed a fix in #6039, which I believe is good to have, as you can't expect every user to execute with `python -B` or set `PYTHONDONTWRITEBYTECODE` while running tests.
